lib/bio/bio.c

```diff
@@ -75,22 +75,34 @@ static ssize_t bio_default_read(struct bdev *dev, void *_buf, off_t offset, size
     }
 
     LTRACEF("buf %p, block %u, len %zd\n", buf, block, len);
+
+    // If the device requires alignment AND our buffer is not alread aligned.
+    bool requires_alignment =
+        (dev->flags & BIO_FLAG_CACHE_ALIGNED_READS) &&
+        (IS_ALIGNED((size_t)buf, CACHE_LINE) == false);
     /* handle middle blocks */
-    if (len >= dev->block_size) {
-        /* do the middle reads */
-        size_t block_count = len / dev->block_size;
-        err = bio_read_block(dev, buf, block, block_count);
+    if (requires_alignment) {
+        while (len >= dev->block_size) {
+            /* do the middle reads */
+            err = bio_read_block(dev, temp, block, 1);
+            memcpy(buf, temp, dev->block_size);
+            if (err < 0)
```

Comment:
Actually on second thought we should bail if `err != dev->block_size` ... using that instead